lib/Backends/Interpreter/Interpreter.cpp

```diff
@@ -772,89 +782,14 @@ static bool quantizeFloatBias(Function *F, FullyConnectedNode &fullyConnected) {
   return true;
 }
 
-/// Channelwise quantize the given float \p bias as int32 using \p inputScale,
-/// per-channel \p scales and \p offsets. \returns false if the bias was already
-/// quantized and thus no change was made and true otherwise.
-static bool channelwiseQuantizeFloatBias(
```

Comment:
@jackm321 But this is the same logic to quantize the bias used in `Graph.cpp` when creating the node when provided with float bias and requested to quantize to Int32QTy. Do you plan to change the logic for the Interpreter? If not then this logic is duplicated.
Other backends can of course opt to provide an already quantized bias when creating the node with their custom logic.

Reply:
@mciprian13 what I'm saying is not moving this logic to graph.cpp but instead keeping it here (or rather in some helper function that can be called here and in other backends that need quantized bias), that way the person who is using loading the model into Glow can load it with quantized or unquantized biases and each backend will do the right thing for what they need. This is because some backends have their own quantization logic for the bias and so to support those backends, the model would have to have biases prequantized with their logic ahead of time.

Reply:
@jackm321 Ok, I will bring back the logic used for the Interpreter in order for you to customize the bias quantization in Interpreter the way you want. But I will not remove anything from the node creator in `Graph.cpp` since currently has the flexibility of:
- being provided with float bias and apply a **default** quantization logic
- being provided directly with quantized bias and so it will do nothing with respect to quantization (and so other backends can do this in order to quantize the bias the way the want). The Interpreter will go through this path after I revert.
Does this sound ok to you?

Reply:
@mciprian13 maybe I'm not understanding but the thing I think is missing (or got removed) from Graph.cpp is the ability to pass in a FloatTy bias and have it _not_ get quantized right there and instead just get attached to the node as a float tensor so that the backend can quantize it instead.

Reply:
@jackm321 
I'm sorry for being wooden-headed but only now I understood you requirements.
So this is what I did:
- I modified the `createChannelwiseQuantizedConv` to have two additional boolean flags: `quantizeFilter` and `quantizeBias` with which you can express if the function should actually quantize the filter/bias or not. If not, the filter/bias will be attached to the node without being touched/quantized such that the backends can do it themselves using their own custom logic.
- In all the importers (Caffe2, ONNX, Pytorch) I used the flag `quantizeBias` to `false` such that the bias operand is not touched at all (allowing therefore to be float).
- I restored the Interpreter logic for manually quantizing the bias.

Does this sound ok to you now?
Thanks!

Reply:
No worries at all, thank you for making those changes so that it works as before, really appreciate it!